cider/exec/template/CaseIR.cpp

```diff
@@ -178,18 +178,23 @@ std::unique_ptr<CodegenColValues> CodeGenerator::codegenCaseExpr(
   Executor::FetchCacheAnchor anchor(cgen_state_);
   const auto& expr_pair_list = case_expr->get_expr_pair_list();
   std::vector<llvm::Value*> then_lvs;
+  std::vector<llvm::Value*> then_null_lvs;
   std::vector<llvm::BasicBlock*> then_bbs;
-  llvm::Value* null_value = nullptr;
   const auto end_bb = llvm::BasicBlock::Create(
       cgen_state_->context_, "end_case", cgen_state_->current_func_);
   for (const auto& expr_pair : expr_pair_list) {
     Executor::FetchCacheAnchor branch_anchor(cgen_state_);
     const auto if_expr_ptr = codegen(expr_pair.first.get(), co, true);
     auto if_expr = dynamic_cast<FixedSizeColValues*>(if_expr_ptr.get());
-    if (if_expr && if_expr->getNull() != nullptr) {
-      null_value = if_expr->getNull();
-    }
+    CHECK(if_expr);
     const auto when_lv = toBool(if_expr->getValue());
+    const auto when_null_lv = if_expr->getNull();
+    CHECK(when_lv);
+    CHECK(when_null_lv);
+    llvm::Value* filter_lv = cgen_state_->llBool(true);
+    filter_lv = cgen_state_->ir_builder_.CreateAnd(
+        filter_lv, cgen_state_->ir_builder_.CreateNot(when_null_lv));
+    auto final_when_lv = cgen_state_->ir_builder_.CreateAnd(filter_lv, when_lv);
```

Comment:
Could you paste the generated IR of this part?

Reply:
%0 = call i8* @cider_ColDecoder_extractArrowBuffersAt(i8* %col_buf0, i64 1)
  %1 = call i8* @cider_ColDecoder_extractArrowBuffersAt(i8* %col_buf0, i64 0)
  %2 = call i64 @fixed_width_int_decode(i8* %0, i32 4, i64 %pos)
  %3 = trunc i64 %2 to i32
  %4 = call i1 @check_bit_vector_clear(i8* %1, i64 %pos)
  %5 = icmp slt i32 %3, %arg_literal_0
  %6 = xor i1 %4, true
  %7 = and i1 true, %6
  %8 = and i1 %7, %5
  br i1 %8, label %then, label %else

Reply:
Like this, do the filter.